src/bodies.cpp

```diff
@@ -106,9 +106,14 @@ struct interscOrder
 }  // namespace detail
 }  // namespace bodies
 
-void bodies::Body::setDimensions(const shapes::Shape* shape)
+void bodies::Body::setDimensionsDirty(const shapes::Shape* shape)
```

Comment:
This function should be moved as inline into the header. It just provides another name for useDimensions().
Maybe you can even use [aliasing](https://www.fluentcpp.com/2017/10/27/function-aliases-cpp/)?

Reply:
I used inlining. The aliasing didn't seem to work for me for some reason.